apps/admin-x-design-system/src/global/form/FileUpload.tsx

```diff
@@ -51,7 +51,7 @@ const FileUpload: React.FC<FileUploadProps> = ({id, onUpload, children, style, u
 
     return (
         <label className={clsx('relative', className)} htmlFor={id} style={style} onDragEnter={handleDragging} onDragLeave={handleStopDragging} onDragOver={handleDragging} onDrop={handleDrop} {...props}>
-            {isDragging && <div className={clsx('absolute inset-1 rounded border-2 border-dashed border-grey-400/25', dragIndicatorClassName)} />}
+            <div className={clsx({'absolute inset-1 rounded': true, 'border-2 border-dashed border-grey-400/25': isDragging, dragIndicatorClassName: isDragging})} />
```

Comment:
```suggestion
            <div className={clsx({'absolute inset-1 rounded': true, 'border-2 border-dashed border-grey-400/25': isDragging, [dragIndicatorClassName]: isDragging})} />
```

I might be misreading this but it looks like `dragIndicatorClassName` got unintentionally converted from a variable to a literal

Reply:
I think we're getting into trouble here because it can be undefined.

Reply:
_sigh_ You can do this in that case:

```suggestion
            <div className={clsx({'absolute inset-1 rounded': true, 'border-2 border-dashed border-grey-400/25': isDragging, ...dragIndicatorClassName && {[dragIndicatorClassName]: isDragging}})} />
```

Reply:
Or... something a lot simpler. Apparently clsx supports multiple arguments.

```suggestion
            <div className={clsx({'absolute inset-1 rounded': true, 'border-2 border-dashed border-grey-400/25': isDragging}, isDragging && dragIndicatorClassName)} />
```

Reply:
I think one of us is confused (very likely me).  I'm seeing dragIndicatorClassName getting passed in like:
```
                                <ImageUpload
                                    deleteButtonClassName='md:invisible absolute pr-3 -right-2 -top-2 flex h-8 w-10 cursor-pointer items-center justify-end rounded-full bg-[rgba(0,0,0,0.75)] text-white group-hover:!visible'
                                    deleteButtonContent={<Icon colorClass='text-white' name='trash' size='sm' />}
                                    editButtonClassName='md:invisible absolute right-[22px] -top-2 flex h-8 w-8 cursor-pointer items-center justify-center text-white group-hover:!visible z-20'
                                    fileUploadClassName='rounded-full bg-black flex items-center justify-center opacity-80 transition hover:opacity-100 -ml-2 cursor-pointer h-[80px] w-[80px]'
                                    fileUploadProps={{dragIndicatorClassName: 'rounded-full'}}
                                    ```

Reply:
What you're doing in [7bd16a1](https://github.com/TryGhost/Ghost/pull/21687/commits/7bd16a13c0f1252db0848e1ca856497ada086e8a) is functionally doing the same thing my suggestion is, just a bit more verbose!

Before the commit, you were doing something like this (rewritten to be more obvious):

```jsx
function component(props) {
	const dragIndicatorClassName = props.dragIndicatorClassName;
    const class = clsx({'dragIndicatorClassName': true});
	// reality: no quotes, js implicitly adds quotes for properties (and Python implicitly expects a variable 🙃)
	// expected:          [dragIndicatorClassName] to tell JS it's a variable (as you did)
	return <div className={classes}>Hallo</div>
}

```

With your commit, you should be able to keep destructuring `dragIndicatorClassName` since you also fixed that ^

Reply:
Thanks, @vikaspotluri123 !  I did some tidying up of the mess I made trying to fix my error, and I think it's a bit cleaner now.  Manual testing says it's still fixed, on windows/chrome, and tests are passing. :)  I'm going to mark this resolved, unless you see anything else amiss!

Reply:
It's still perfectly fine, but you should be able to micro-optimize it by using a string instead of array!

`isDragging && [dragIndicatorClassName]`
-->
`isDragging && dragIndicatorClassName`